tasklist/els-schema/src/main/java/io/camunda/tasklist/schema/IndexMapping.java

```diff
@@ -59,7 +63,37 @@ public Map<String, Object> toMap() {
     return properties.stream()
         .collect(
             Collectors.toMap(
-                IndexMappingProperty::getName, IndexMappingProperty::getTypeDefinition));
+                IndexMappingProperty::getName,
+                property -> convertToOrderedStructures(property.getTypeDefinition())));
+  }
+
+  /**
+   * On Opensearch Schema validation fails when the schema is created by the Exporter (primarility
+   * for tasklist-task index and the `join` field), due to the order of parsed index mapping
+   * properties. Using TreeMap and sorting the Lists present in the properties ensures the order of
+   * properties when comparing the existing mappings with the ones defined in the schema.
+   */
+  private Object convertToOrderedStructures(final Object value) {
```

Comment:
❓ Stupid question but couldn't we use ObjectMapper and JsonObject to compare against?

Reply:
The comparison of the indices is currently based on `com.google.common.collect.MapDifference` to perform a diff between HashMaps. However, for nested Maps the order of keys and for lists inside them need to be similar.

There could be other ways to achieve this fix as well. This is what came to my mind to unblock this issue. 

Also, changing the order of the entries in the JSON schema fixed this issue but could arise sooner or later. It's weird it's not happening on the CamundaExporter aswell because the diff is done with the exact same way :dizzy: